Homework6/pingpong.cpp

```diff
@@ -0,0 +1,101 @@
+#include <iostream>
+#include <chrono>
+#include <atomic>
+#include <mutex>
+#include <thread>
+#include <condition_variable>
+
+using namespace std;
+
+constexpr chrono::milliseconds flightTime = 300ms;
+constexpr chrono::milliseconds clockTick = 200ms;
+
+enum gameState {
+    FINISH = 0, PING, PONG
+};
+
+class PingPong {
+    mutex ballLock;
+
+    atomic<int> counter;
+    atomic<int> turnNumber = 1;
+    atomic<bool> timerOut = false;
+
+    atomic<gameState> status = gameState::PING;
+
+    condition_variable condition;
+public:
+
+    explicit PingPong(int turns) : counter(turns) {}
+
+    void ping() {
+        while (counter && !timerOut) {
+            unique_lock<mutex> lock(ballLock);
+            condition.wait_for(lock, clockTick);
```

Comment:
użycie wait_for tutaj nie jest w 100% bezpieczne. Może być tak, że clockTick upłynie i tutaj nawet nie sprawdzasz czy timeout nastąpił, czy był notify. 
To co masz w if dodaj po prostu jako predykat do wait_for, a może nawet zrób z niego sam wait?

Reply:
done